src/therapy/database/dynamodb.py

```diff
@@ -0,0 +1,635 @@
+        therapy aliases, this method won't pick it up. It just wants to see if a few
+        critical tables have at least a small number of records.
+
+        :return: True if queries successful, false if DB appears empty
+        """
+        sources = self.therapies.query(
+            IndexName="item_type_index",
+            KeyConditionExpression=Key("item_type").eq("source"),
+        ).get("Items", [])
+        if len(sources) < len(SourceName):
+            _logger.info("Therapy sources table is missing expected sources.")
+            return False
+
+        records = self.therapies.query(
+            IndexName="item_type_index",
+            KeyConditionExpression=Key("item_type").eq("identity"),
+            Limit=1,
+        )
+        if len(records.get("Items", [])) < 1:
+            _logger.info("Therapy records index is empty.")
+            return False
+
+        normalized_records = self.therapies.query(
+            IndexName="item_type_index",
+            KeyConditionExpression=Key("item_type").eq(RecordType.MERGER.value),
+            Limit=1,
+        )
+        if len(normalized_records.get("Items", [])) < 1:
+            _logger.info("Normalized therapy records index is empty.")
+            return False
+
+        return True
+
+    def initialize_db(self) -> None:
+        """Create therapy_normalizer table if not already created."""
+        if not self.check_schema_initialized():
+            self._create_therapies_table()
+
+    def get_source_metadata(self, src_name: Union[str, SourceName]) -> Optional[Dict]:
+        """Get license, versioning, data lookup, etc information for a source.
+
+        :param src_name: name of the source to get data for
+        :return: Dict containing metadata if lookup is successful
+        """
+        if isinstance(src_name, SourceName):
+            src_name = src_name.value
+        if src_name in self._cached_sources:
+            return self._cached_sources[src_name]
+        else:
+            pk = f"{src_name.lower()}##source"
+            concept_id = f"source:{src_name.lower()}"
+            metadata = self.therapies.get_item(
+                Key={"label_and_type": pk, "concept_id": concept_id}
+            ).get("Item")
+            if not metadata:
+                return None
+            self._cached_sources[src_name] = metadata
+            return metadata
+
+    def get_record_by_id(
+        self, concept_id: str, case_sensitive: bool = True, merge: bool = False
+    ) -> Optional[Dict]:
+        """Fetch record corresponding to provided concept ID
+
+        :param str concept_id: concept ID for therapy record
+        :param bool case_sensitive: if true, performs exact lookup, which is more
+            efficient. Otherwise, performs filter operation, which doesn't require
+            correct casing.
+        :param bool merge: if true, look for merged record; look for identity record
+            otherwise.
```

Comment:
```suggestion
        :param concept_id: concept ID for therapy record
        :param case_sensitive: if true, performs exact lookup, which is more
            efficient. Otherwise, performs filter operation, which doesn't require
            correct casing.
        :param merge: if true, look for merged record; look for identity record
            otherwise.
```